### ISSUE TITLE
Throttling seems to be disabled despite the claims in readme

### ISSUE THREAD

**Issue by postatum** [octokit/octokit.js#2168]
### Please avoid duplicates

- [X] I checked [all open bugs](https://github.com/octokit/octokit.js/issues?q=is%3Aissue+is%3Aopen+label%3Abug) and none of them matched my problem.

### Reproducible test case
```js
const { Octokit } = require('octokit')
const { throttling } = require("@octokit/plugin-throttling")
Octokit.plugins.includes(throttling)
```

### Please select the environment(s) that are relevant to your bug report

- [ ] TypeScript
- [ ] Enterprise
- [ ] Browsers
- [X] Node
- [ ] Deno

### Version

octokit@1.7.0

### What happened?

Hi.
Note: I'm not sure if this is a documentation issue or really a bug.

The README mentions throttling in these pieces:

> Octokit implements request throttling using @octokit/plugin-throttling
> By default, requests are retried once and warnings are logged in case hitting a rate or abuse limit

> In both cases a given request is authenticated, retried, and throttled transparently by the octokit instance which also manages the accept and user-agent headers as needed.

Overall these seem to say "throttling is enabled by default". Though, the first piece is less clear about that than the second one.

Despite the above, throttling seems to be disabled in code: https://github.com/octokit/octokit.js/blob/v1.7.0/src/octokit.ts#L13

PS. I started researching this after I hit rate limit and throttling didn't seem to work.

### Would you be interested in contributing a fix?

- [ ] yes

**Comment by Chocrates** [comment]
Hey @postatum I think this would be a documentation bug, throttling is handled in https://github.com/octokit/plugin-throttling.js

Sample of setting it up

```
const { throttling } = require("@octokit/plugin-throttling");
  const { retry } = require("@octokit/plugin-retry");
  const { Octokit } = require("@octokit/rest");
  const MyOctokit = Octokit.plugin(throttling).plugin(retry);

const client = new MyOctokit({
          auth: "token " + token,
          throttle: {
              onRateLimit: (retryAfter, options) => {
                  console.warn(
                      `Request quota exhausted for request ${options.method} ${options.url}`
                  );
                  console.log(
                      `Retrying after ${retryAfter} seconds! Retry Count: ${options.request.ret  ryCount}`
                  );
                  return true;
              },
>>            onAbuseLimit: (retryAfter, options) => {
                  console.warn(
                      `Abuse detected for request ${options.method} ${options.url}`
                  );
              },
≃         }, request: { retries: 1, doNotRetry: [403], },
      });
```

**Comment by Chocrates** [comment]
Additionally I don't think I agree that the docs are invalid, they mention that requests will retry once, code for that is here https://github.com/octokit/octokit.js/blob/v1.7.0/src/octokit.ts#L41

But that throttling needs to be done with the throttling plugin which is not included by default as you mentioned in the opening ticket.

**Comment by postatum** [comment]
I see. Thanks for the answer!

**Comment by Fs00** [comment]
> Additionally I don't think I agree that the docs are invalid, they mention that requests will retry once, code for that is here https://github.com/octokit/octokit.js/blob/v1.7.0/src/octokit.ts#L41

I've stumbled upon this recently and, after investigating a bit, it seems like that the code @Chocrates is referring to is not actually used.
The `throttle.onRateLimit` and `throttle.onAbuseLimit` options are being [used by the throttling plugin here](https://github.com/octokit/plugin-throttling.js/blob/master/src/index.ts#L81), and therefore if the throttling plugin is disabled, those functions will never be called.

Also, it seems like that the throttling plugin was disabled by mistake in #2116. That PR bumps the version of the throttling plugin to include the fix for #2115, but it also strangely disables the plugin without any notice.
Wouldn't it be best to re-enable it in order to fix this issue?

**Comment by gr2m** [comment]
It's been a while, but it looks like the TypeScript problems no longer persist, so I re-enabled the throttling plugin in https://github.com/octokit/octokit.js/pull/2224
